lab-anthony/test/auth-router-test.js

```diff
@@ -0,0 +1,168 @@
+  before((done)=>{
+    debug('before module auth-router');
+    if(!server.isRunning){
+      server.listen(port, ()=>{
+        server.isRunning = true;
+        debug('server up on port', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done)=>{
+    debug('after module auth-router');
+    if(server.isRunning){
+      server.close(()=>{
+        server.isRunning = false;
+        debug('server is down');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing successful POST /api/signup', function(){
+    after((done)=>{
+      debug('after POST /api/signup');
+      userController.removeAllUsers()
+      .then(()=> {
+        done();
+      })
+      .catch(()=>{
+        done();
+      });
+    });
+
+    it('should return a token', function(done){
+      debug('test POST /api/signup');
+      request.post(`${baseURL}/signup`)
+      .send({
+        username: 'Anthony',
+        password: 'ballin'
+      })
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.text.length).to.equal(205);
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('testing failed POST /api/signup', function(){
+    it('should return 400 bad request', function(done){
+      debug('test POST /api/signup');
+      request.post(`${baseURL}/signup`)
+      .send({})
+      .then(done)
+      .catch(err => {
+        try {
+          expect(err.response.status).to.equal(400);
+          expect(err.response.text).to.equal('BadRequestError');
+          done();
+        } catch(err) {
+          done(err);
+        }
+      });
+    });
+  });
+
+  //testing signin
+  describe('testing successful POST /api/signin', function(){
+    before((done)=>{
+      debug('before module signin');
+      authController.signup({username: 'Anthony', password: 'ballin'})
+      .then(()=> done())
+      .catch(done);
+    });
+
+    after((done)=>{
+      debug('after POST /api/signup');
+      userController.removeAllUsers()
+      .then(()=> {
+        done();
+      })
+      .catch(()=>{
+        done();
+      });
+    });
+
+    it('should return a token', function(done){
+      debug('test POST /api/signup');
+      request.get(`${baseURL}/signin`)
+      .auth('Anthony', 'ballin')
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.text.length).to.equal(205);
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('testing failed GET /api/signin', function(){
+    before((done)=>{
+      debug('before module signin');
+      authController.signup({username: 'Anthony', password: 'ballin'})
+      .then(()=> done())
+      .catch(done);
+    });
+
+    after((done)=>{
+      debug('after GET /api/signin');
+      userController.removeAllUsers()
+      .then(()=> {
+        done();
+      })
+      .catch(()=>{
+        done();
+      });
+    });
+
+    it('should return 401', function(done){
```

Comment:
You should probably explain the reason why you expect a 401. If there were multiple situations you were testing that could all give 401s, you would want to be specific about why it should be giving you a 401 in each situation.